include/RemotePlugin.h

```diff
@@ -143,27 +147,26 @@ class LMMS_EXPORT RemotePlugin : public QObject, public RemotePluginBase
 		m_commMutex.unlock();
 	}
 
-	PluginPinConnector& pinConnector()
-	{
-		return m_pinConnector;
-	}
-
 public slots:
 	virtual void showUI();
 	virtual void hideUI();
 
 protected:
-	inline void setSplittedChannels( bool _on )
-	{
-		m_splitChannels = _on;
-	}
+	bool m_failed;
 
+	//! Signal to derived classes
+	virtual void bufferUpdated() {}
 
-	bool m_failed;
-private:
-	void resizeSharedProcessingMemory();
+	auto frames() const -> f_cnt_t { return m_frames; }
+	auto channelsIn() const -> pi_ch_t { return m_channelsIn; }
+	auto channelsOut() const -> pi_ch_t { return m_channelsOut; }
 
+	auto inputBuffer() const -> Span<float> { return m_inputBuffer; }
+	auto outputBuffer() const -> Span<float> { return m_outputBuffer; }
 
+	PluginPinConnector* m_pinConnector = nullptr;
```

Comment:
I'm not aware of the C++ Core Guidelines recommending const member variables in cases like this, but if you point to where it says that, I can change it.

Whenever I use a raw pointer, you can assume it is non-owning. If for some reason I used an owning raw pointer, I would explicitly note that it is owning, otherwise I made a mistake.

Reply:
The [guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable) say "By default, make objects immutable". They mention exceptions, but class members (like in this line) are not mentioned: "Flag non-const variables that are not modified (except for parameters to avoid many false positives and returned local variables)". So I would assume that the pointer should be const. However, you could argue, that like in the "parameters" case, this confuses the reader, making them think the pointed-to memory is immutable.

> Whenever I use a raw pointer, you can assume it is non-owning. If for some reason I used an owning raw pointer, I would explicitly note that it is owning

:+1:

Reply:
Done in https://github.com/LMMS/lmms/pull/7459/commits/3661bbdcc709f3612dd9615d98120326876e3f24